### ISSUE TITLE
Error profiling throughput grid for Skyplane on AWS

### ISSUE THREAD

**Issue by parasj** [skyplane-project/skyplane#596]
From #595 

Only profiling AWS: `skyplane experiments throughput-grid --no-enable-azure --no-enable-gcp --no-enable-gcp-standard`

Error: 
```
19:40:55 [INFO]  Provisioning AWS instances in ['eu-north-1', 'ap-south-1', 'eu-west-3', 'eu-west-2', 'eu-west-1', 'ap-northeast-3', 'ap-northeast-2', 'ap-northeast-1', 
'sa-east-1', 'ca-central-1', 'ap-southeast-1', 'ap-southeast-2', 'eu-central-1', 'us-east-1', 'us-east-2', 'us-west-1', 'us-west-2']
⠦ Add IP to aws security groups ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  0/17 0:00:0219:40:58 [ERROR] Error running add_ips_to_security_group: 'NoneType' object is not iterable
⠇ Add IP to aws security groups ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  0/17 0:00:0319:40:58 [ERROR] Error running add_ips_to_security_group: 'NoneType' object is not iterable
⠹ Add IP to aws security groups ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  0/17 0:00:0319:40:58 [ERROR] Error running add_ips_to_security_group: 'NoneType' object is not iterable
19:40:58 [ERROR] Error running add_ips_to_security_group: 'NoneType' object is not iterable
19:40:58 [ERROR] Error running add_ips_to_security_group: 'NoneType' object is not iterable
...
```

**Comment by lynnliu030** [comment]
The error can be fixed by adding additional check for the case where`ips is None`. 

Then I was getting similar error as in #598 during the server [setup ](https://github.com/skyplane-project/skyplane/blob/main/skyplane/cli/experiments/cli_profile.py#L199)phase on AWS instances. [Here](https://github.com/skyplane-project/skyplane/blob/main/skyplane/cli/experiments/cli_profile.py#L203), might need to change to `sudo yum` for installing packages including `debconf`. 

But `yum update` and the installation process somehow takes forever to finish, seems like it just gets stuck

**Comment by parasj** [comment]
@lynnliu030 Can you draft a PR to fix this?

For the `deb vs apt` setup, I would recommend adding functionality to override the AMI. On all clouds, we should install Ubuntu for profiling versus AWS Bottlerocket or Google Container OS on VMs (which we use for security reasons on gateways). In `provision_instance` (e.g. aws_cloud_provider.py:326), add an `os: str` argument that defaults to `bottlerocket` on AWS and `googlecos` on GCP. If it's `ubuntu`, then select the Ubuntu AMI using resolve:ssm which automatically resolves the correct image in every AWS region. Similarly do so for GCP as well.

**Comment by lynnliu030** [comment]
I changed the AMI for profiling, there're some following errors I was able to solve and test the AWS connections successfully. One of the blocking error is I got is a time-out error when testing the GCP to GCP connections. 
```
(myenv) ~/Documents/Projects/skyplane$ skyplane experiments throughput-grid --no-enable-aws --no-enable-azure
11:47:48 [INFO]  Provisioning GCP instances in ['us-east1-b', 'us-east1-c']
11:47:50 [INFO]  (GCP) provisioning missing regions: ['us-east1-b']
Image:  projects/ubuntu-os-cloud/global/images/family/ubuntu-1804-lts
11:48:42 [INFO]  Provisioning GCP instances in ['us-east1-b', 'us-east1-c']
11:48:44 [INFO]  (GCP) provisioning missing regions: ['us-east1-b']
Image:  projects/ubuntu-os-cloud/global/images/family/ubuntu-1804-lts

Experiment configuration: (total pairs = 2)
        Group 0: (1 items)
        gcp:us-east1-b:STANDARD -> gcp:us-east1-b:PREMIUM
        Group 1: (1 items)
        gcp:us-east1-b:PREMIUM -> gcp:us-east1-b:STANDARD

iperf_runtime=5, iperf3_connections=64
Approximate runtime: 30s (assuming 10s startup time)
Approximate data to send: 6.25GB (assuming 5Gbps)
Approximate cost: $0.62 (assuming $0.10/GB)
11:49:56 [DEBUG] Experiment tag: 2022.10.11_18.49_posh-semidarkness-tawny-stardom_5s_64c
11:49:56 [DEBUG] Log directory: /Users/shuL/Documents/Projects/skyplane/data/logs/throughput_grid/2022.10.11_18.49_posh-semidarkness-tawny-stardom_5s_64c

Run experiment? (tag: 2022.10.11_18.49_posh-semidarkness-tawny-stardom_5s_64c) [Y/n]: y
11:49:58 [DEBUG] Raw iperf3 log directory: 
/Users/shuL/Documents/Projects/skyplane/data/logs/throughput_grid/2022.10.11_18.49_posh-semidarkness-tawny-stardom_5s_64c/raw_iperf3_logs
⠙ Parallel eval group 0 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0/1 0:02:1011:52:09 [ERROR] Exception: 'sum_sent'
Traceback (most recent call last):
  File "/Users/shuL/Documents/Projects/skyplane/skyplane/cli/experiments/cli_profile.py", line 69, in start_iperf3_client
    out_rec["throughput_sent"] = result["end"]["sum_sent"]["bits_per_second"]
KeyError: 'sum_sent'
⠙ Parallel eval group 0 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0/1 0:02:1011:52:09 [ERROR] (gcp:us-east1-b -> gcp:us-east1-b) iperf3 client failed: {
        "start":        {
                "connected":    [],
                "version":      "iperf 3.1.3",
                "system_info":  "Linux skyplane-gcp-78e94bd79280407bb7722b1e3a729c81 5.4.0-1089-gcp #97~18.04.1-Ubuntu SMP Wed Sep 28 03:32:39 UTC 2022 x86_64"
        },
        "intervals":    [],
        "end":  {
        },
        "error":        "error - unable to connect to server: Connection timed out"
}
 
gcp:us-east1-b:STANDARD_gcp:us-east1-b:PREMIUM: 0.00Gbps
```

Any clues on what might be the problems? I tried `skyplane ssh` to connect to one of the instance and try iperf3 there, I get similar error. Note that when I ssh to the instance I need to enter passphrase `skyplane` for the connection; shall we do something like including the key password inline for ssh?

**Comment by parasj** [comment]
Hey @lynnliu030, is this fixed by adding private IPs to the firewall?

**Comment by lynnliu030** [comment]
@parasj I tried but I don't think this can be solved by adding private IPs to fw. My fix in #605 is to add iperf default port `5201` to [fw_body](https://github.com/skyplane-project/skyplane/blob/d6315b442aae9ef7acc4f40cb6b497d21f52a50d/skyplane/compute/gcp/gcp_cloud_provider.py#L176-L209) when configuring skyplane firewall for GCP 
```
fw_body = {
            "name": "skyplanessh",
            "network": "global/networks/skyplane",
            "allowed": [
                {"IPProtocol": "tcp", "ports": ["22", "5201"]},
                {"IPProtocol": "udp", "ports": ["1-65535"]},
                {"IPProtocol": "icmp"},
            ],
            "description": "Allow all traffic from all IPs",
            "sourceRanges": [ip],
        }
````
